WordPress/gutenberg
# Remove the navigation edit button because it leads to a useless screen

## What?


The navigation block exposes an edit button in its toolbar. Clicking that leads the user to a screen where they can edit the navigation contents.  Let's remove it because it's useless.

## Why?


Editing navigation in isolation never amounted to much and that button creates the illusion that the navigation is not editable in place already. Since we have some discoverability issues around editing menus using the navigation block we don't want any more confusion on top.


## How?


Just removed the hook that added the button via a block edit filter.

## Testing Instructions





1. Add a navigaiton block
2. Select the block
3. **Notice there is no edit button in the block toolbar**

### Testing Instructions for Keyboard


## Screenshots or screencast 

### Before


![Screenshot 2024-02-20 at 16 25 57](https://github.com/WordPress/gutenberg/assets/107534/6b8de25a-aa80-404b-b3b3-66003569757c)


### After

![Screenshot 2024-02-20 at 16 21 28](https://github.com/WordPress/gutenberg/assets/107534/3dd22f63-3204-426c-925a-bac3ba3a166b)
